cli/support.go

```diff
@@ -0,0 +1,202 @@
+			r.InitClient(client),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			var (
+				log = slog.Make(sloghuman.Sink(inv.Stderr)).
+					Leveled(slog.LevelDebug)
+				deps = support.Deps{
+					Client: client,
+					Log:    log,
+				}
+			)
+
+			if len(inv.Args) == 0 {
+				return xerrors.Errorf("must specify workspace name")
+			}
+			ws, err := namedWorkspace(inv.Context(), client, inv.Args[0])
+			if err != nil {
+				return err
+			}
+
+			deps.WorkspaceID = ws.ID
+
+			agentName := ""
+			if len(inv.Args) > 1 {
+				agentName = inv.Args[1]
+			}
+
+			agt, found := findAgent(agentName, ws.LatestBuild.Resources)
+			if !found {
+				return xerrors.Errorf("could not find agent named %q for workspace", agentName)
+			}
+
+			deps.AgentID = agt.ID
+
+			bun, err := support.Run(inv.Context(), &deps)
+			if err != nil {
+				return err
+			}
+
+			if outputPath == "" {
+				cwd, err := filepath.Abs(".")
+				if err != nil {
+					return xerrors.Errorf("could not determine current working directory: %w", err)
+				}
+				loc, err := tz.TimezoneIANA()
+				if err != nil {
+					loc = time.UTC
+				}
+				tsStr := time.Now().In(loc).Format("2006-01-02-150405")
+				fname := "coder-support-" + tsStr + ".zip"
+				outputPath = filepath.Join(cwd, fname)
+			}
+
+			w, err := os.Create(outputPath)
+			if err != nil {
+				return xerrors.Errorf("create output file: %w", err)
+			}
+			zwr := zip.NewWriter(w)
+			defer zwr.Close()
+			if err := writeBundle(bun, zwr); err != nil {
+				return xerrors.Errorf("write support bundle to %s: %w", outputPath, err)
+			}
+			return nil
+		},
+		Hidden: true, // TODO: un-hide
+	}
+	cmd.Options = clibase.OptionSet{
+		{
+			Flag:          "output",
+			FlagShorthand: "o",
+			Env:           "CODER_SUPPORT_BUNDLE_OUTPUT",
+			Description:   "Path to which to output the generated support bundle. Defaults to coder-support-YYYYmmdd-HHMMSS.zip.",
+			Value:         clibase.StringOf(&outputPath),
+		},
+	}
+
+	return cmd
+}
+
+func findAgent(agentName string, haystack []codersdk.WorkspaceResource) (*codersdk.WorkspaceAgent, bool) {
+	for _, res := range haystack {
+		for _, agt := range res.Agents {
+			if agentName == "" {
+				// just return the first
+				return &agt, true
+			}
+			if agt.Name == agentName {
+				return &agt, true
+			}
+		}
+	}
+	return nil, false
+}
+
+func writeBundle(src *support.Bundle, dest *zip.Writer) error {
+	for k, v := range map[string]any{
+		"deployment/buildinfo.json":    src.Deployment.BuildInfo,
+		"deployment/config.json":       src.Deployment.Config,
+		"deployment/experiments.json":  src.Deployment.Experiments,
+		"deployment/health.json":       src.Deployment.HealthReport,
+		"network/netcheck_local.json":  src.Network.NetcheckLocal,
+		"network/netcheck_remote.json": src.Network.NetcheckRemote,
+		"workspace/workspace.json":     src.Workspace.Workspace,
+		"workspace/agent.json":         src.Workspace.Agent,
+	} {
+		f, err := dest.Create(k)
```

Comment:
Does this need to be closed? (Didn't check)

Reply:
No it's just an `io.Writer`. You close the entire `zip.Writer` when you're done.